### PR TITLE
Adding MathML Core support to rsx

### PR DESCRIPTION
**Pull request by Ameyanagi**
This PR adds a [MathML Core](https://w3c.github.io/mathml-core/#mathml-elements-and-attributes) support to the rsx as discussed in issue https://github.com/DioxusLabs/dioxus/issues/1778.

Deprecated tags and attributes are not included in the code base.
annotation-xml is aliased to annotationXml due to the naming restrictions of rsx macro.

@stefnotch @Jah-On

### CODE REVIEW

**Review by Jah-On — APPROVED** [review]
Looks good!

**Review by ealmloff — APPROVED** [review]
Thank you!

File: packages/html/src/elements.rs [hunk 1]
```diff
@@ -1744,4 +1744,199 @@ builder_constructors! {
...
+    munderover "http://www.w3.org/1998/Math/MathML" {
+        accent: Bool DEFAULT,
+        accentunder: Bool DEFAULT,
+    };
+
+    /// Build a
+    /// [`<semantics>`](https://developer.mozilla.org/en-US/docs/Web/MathML/Element/semantics)
+    /// element.
+    semantics "http://www.w3.org/1998/Math/MathML" {
+        encoding: String DEFAULT,
```
Comment (stefnotch) [inline packages/html/src/elements.rs:1940]: The semantics element doesn't accept an encoding

**Comment by stefnotch** [comment]
Will we need a MathML equivalent of https://github.com/DioxusLabs/dioxus/blob/53727b344de9af5bcd160fb99703a09ec0da2da4/packages/html/src/global_attributes.rs#L1643 ? The MathML core spec specifies a few global attributes.

File: packages/html/src/elements.rs (hunk 1 above)
Comment (ealmloff) [reply packages/html/src/elements.rs:1940]: It looks like it does:
- https://developer.mozilla.org/en-US/docs/Web/MathML/Element/semantics#encoding
Screenshot 2024-01-19 at 8 01 12 AM

**Comment by Ameyanagi** [comment]
@stefnotch @ealmloff 
I will try to add the global attributes.

File: packages/html/src/elements.rs (hunk 1 above)
Comment (stefnotch) [reply packages/html/src/elements.rs:1940]: Huh, interesting. The MathML Core spec seems to not mention it https://w3c.github.io/mathml-core/#semantics-and-presentation

**Pull request merged**